include/oneapi/dpl/experimental/kt/two_pass_scan.h

```diff
@@ -355,6 +359,19 @@ two_pass_scan(sycl::queue q, _InRng&& __in_rng, _OutRng&& __out_rng,
                             carry_last = carry;
                     }
                 }
+                // For the exclusive scan case:
+                // While the first sub-group is doing work, have the last item in the group store the last element
+                // in the block to temporary storage for use in the next block.
+                // This is required to support in-place exclusive scans as the input values will be overwritten.
+                if constexpr (!Inclusive)
+                {
+                    auto global_id = ndi.get_global_linear_id();
+                    if (global_id == num_work_groups * work_group_size - 1)
```

Comment:
Could we create a variable for the expression `num_work_groups * work_group_size` (maybe called `num_work_items`) and use it both here and on line 146 (before creation of `globalRange`)?

Reply:
Yes, I have done this.